### ISSUE TITLE
import/default

### ISSUE THREAD

**Issue by quanru** [benmosher/eslint-plugin-import#892]
it would report a error, when a third-party npm package exposed a index.js like this:

```js
import Swiper from './src/'
module.exports = Swiper
```

but it perform normally when exposed like:

```js
import Swiper from './src/'
export default Swiper
```

or like:

```js
const Swiper = require('./src/')
module.exports = Swiper
```

example package: https://github.com/leecade/react-native-swiper/blob/master/index.js

**Comment by ljharb** [comment]
Third-party packages should be transpiled prior to being published; despite what's common in the react-native ecosystem, publishing `import` or `export` is a bug.

**Comment by benmosher** [comment]
it is reporting correctly: there is no default export in that module. 

you're not the first person to report this, though. seems to be a growing pattern. I don't agree with it philosophically, FWIW. 

though that probably means I'll love it and use it exclusively in 6-9 months 😅

if you could provide a reference document / blog post / anything for this pattern I'd love to see it. not sure why packages would be published like this, smells like a hack around Webpack/Babel behaviors. I'm not against supporting it eventually in an isolated way if that's the case, but want to understand it first.
